File: app/src/main/java/me/gabeg/sicksends/boulder/SsBoulderRepository.java
package me.gabeg.sicksends.boulder;

import android.app.Application;

import me.gabeg.sicksends.problem.SsProblemDatabase;
import me.gabeg.sicksends.problem.SsProblemRepository;

/**
 * Repository for boulder problems.
 */
public class SsBoulderRepository
	extends SsProblemRepository<SsBoulderDao, SsBoulder>
{

	/**
	 */
	public SsBoulderRepository(Application app)
	{
		super(app);
	}

	/**
	 * @see SsProblemRepository#buildDao(SsProblemDatabase)
	 */
	@Override
	protected void buildDao(SsProblemDatabase db)
	{
		this.mProblemDao = db.boulderDao();
	}

}

File: app/src/main/java/me/gabeg/sicksends/boulder/SsBoulderRepository.kt
package me.gabeg.sicksends.boulder

import androidx.lifecycle.LiveData
import me.gabeg.sicksends.problem.SsGenericProblemRepository

/**
 * Repository for boulder problems.
 */
class SsBoulderRepository(dao : SsBoulderDao)
	: SsGenericProblemRepository<SsBoulderProblem>(dao)
